api/v1alpha1/hostedcluster_types.go

```diff
@@ -42,7 +42,9 @@ type HostedClusterSpec struct {
 
 	// Services defines metadata about how control plane services are published
 	// in the management cluster.
-	Services []ServicePublishingStrategyMapping `json:"services"`
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default={{service:"APIServer",servicePublishingStrategy:{type:"LoadBalancer"}},{service:"VPN",servicePublishingStrategy:{type:"LoadBalancer"}},{service:"OAuthServer",servicePublishingStrategy:{type:"Route"}}}
+	Services []ServicePublishingStrategyMapping `json:"services,omitempty"`
```

Comment:
just FYI don't want to prevent from merging but if these don't exist I do believe all reconciliation will fail

Reply:
Specifically here:
https://github.com/openshift/hypershift/blob/main/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go#L452

Reply:
well not fail but I think the creation of the critical services like Kube APIServer, Openvpn, and Oauth will not happen if it is empty. I also believe the machine config server will fail here:

Reply:
https://github.com/openshift/hypershift/blob/main/hypershift-operator/controllers/machineconfigserver/machineconfigserver_controller.go#L334

Unless maybe I'm missing the kube builder magic of if it's empty it'll just magically populate

Reply:
It's getting defaulted when empty, but now it makes me think that I don't need to make the fields optional. Will update in the morning.

Reply:
Can we please do the same defaulting for the `hostedControlPlane` `HostedControlPlaneSpec.Services` field?

IMO this field should be required and defaulted as in here. That's just UX.
Then the backend controller should no be doing any assumption, it should handle all input possibilities and so it should fail appropriately when the field does not exist.

Reply:
Agree with @enxebre about making it required but defaulted